src/main/kotlin/calculator/StringCalculator.kt

```diff
@@ -0,0 +1,61 @@
+package calculator
+
+class StringCalculator(private val str: String?) {
+
+    fun calculate(): Int = when (str.isNullOrBlank()) {
+        true -> { RESULT_EMPTY_VALUE }
+        false -> {
+            val matchResult: MatchResult? = Regex(REGEX_DIVIDER).find(str)
+            val divider = getDivider(matchResult)
+
+            require(!divider.isNullOrEmpty()) { "divider" }
+
+            val splitStr = splitStr(divider, matchResult)
+            val numbers = splitToNumbers(splitStr)
+            add(numbers)
+        }
+    }
```

Comment:
when 표현식의 적절한 사용은 가독성을 높이지만 매번 긍적적인 효과를 얻을수는 없어요. 
아래 로직과 작성해주신 로직중 처음 보는 사람이 이해하기 쉬운 코드가 무엇인지 고민해보세요 😄 
```suggestion
    fun calculate(): Int {
        if (str.isNullOrBlank()) {
            return RESULT_EMPTY_VALUE
        }
        
        val matchResult: MatchResult? = Regex(REGEX_DIVIDER).find(str)
        val divider = getDivider(matchResult)

        require(!divider.isNullOrEmpty()) { "divider" }

        val splitStr = splitStr(divider, matchResult)
        val numbers = splitToNumbers(splitStr)
        return add(numbers)
    }
```